### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
+
+The `GIR` is used to generate both sys level and Rust-user API level.
+
+## Generate FFI (sys level)
+
+You first need to get the Gir file corresponding to the project you want to bind. Then when you have it, you need to list object you want to generate in a toml file.
+
+*This part needs more information*
+
+When you're ready, let's generate the FFI part. Don't forget to download the gir files as well (you can get them from [here](https://github.com/gtk-rs/gir-files)):
```
Comment (EPashkin) [inline README.md]: May be need make clearer that downloaded `gtk .gir files`

Comment (GuillaumeGomez) [inline README.md]: I'm still looking for the place where they're supposed to be actually haha.

Comment (GuillaumeGomez) [inline README.md]: I think it'll be enough. I'll just write something a bit easier to understand.

**Comment by GuillaumeGomez** [comment]
Updated,

File: README.md [hunk 2]
```diff
@@ -1,8 +1,27 @@
-## Generate something from GIR
+# GIR
+
+The `GIR` is used to generate both sys level and Rust-user API level.
+
+## Generate FFI (sys level)
+
+You first need to get the Gir file corresponding to the project you want to bind. Then when you have it, you need to list object you want to generate in a toml file.
```
Comment (EPashkin) [inline README.md]: List object need only for next part. Sys generated all

Comment (EPashkin) [inline README.md]: May be need link to https://github.com/gtk-rs/sys/blob/master/conf/gir-gtk.toml as example?

**Comment by GuillaumeGomez** [comment]
Ok, so I reorganized a bit the README. Anything else I should update?

File: README.md [hunk 3]
```diff
@@ -1,8 +1,29 @@
...
+cargo run --release -c YourGirFile.toml -d ../gir-files -m sys -o the-output-directory-sys
 ```
-where `gir-files` contains the [GIR definitions](https://github.com/gkoz/gir-files).
-The generated files will be placed in `gtk`
+
+The generated files will be placed in `the-output-directory-sys`. You now have the sys part of your binding!
+
+## Generate the Rust-user API level
+
+Normally, a toml file corresponding to your crate has been generated (you can take a look to [gir-gtk.toml](https://github.com/gtk-rs/sys/blob/master/conf/gir-gtk.toml) as an example).
```
Comment (EPashkin) [inline README.md]: gir-gtk.toml link belong to ffi part.
here better local `Gir_Gtk.toml` or gtk's Gir.toml

Comment (EPashkin) [inline README.md]: If you prefer local then ffi link can be local too `tests/sys/gir-gtk.toml`

File: README.md [hunk 4]
```diff
@@ -1,8 +1,31 @@
...
+
+When you're ready, let's generate the FFI part. Don't forget to download the gir files as well.
+
+```shell
+cargo run --release -c YourGirFile.toml -d ../gir-files -m sys -o the-output-directory-sys
+```
+
+The generated files will be placed in `the-output-directory-sys`. You now have the sys part of your binding!
+
+A TOML file should have been generated as well (you can take a look to [gir-gtk.toml](https://github.com/gtk-rs/sys/blob/master/conf/gir-gtk.toml) as an example).
```
Comment (EPashkin) [inline README.md]: You wrong here, toml file writed by hand and passed to gir as `YourGirFile.toml`

Comment (EPashkin) [inline README.md]: Hm, 'TOML file` ??? May be you mean `Cargo.toml`? Then you partially right, it generated, if I not wrong

Comment (GuillaumeGomez) [inline README.md]: No, I was speaking about the `Gir.toml` file but I just checked and I'm wrong. Thanks for pointing this out!

**Comment by bcl** [comment]
A higher level discussion of the differences (and need) for the sys level and normal level packages would be helpful. This is part of what confused me at first, it wasn't clear that there were 2 parts. The gudev-rs code was a really good example of what needed to be done.

Some caveats about what it can and cannot generate would also be helpful as would detailed documentation about what the format of Gir.toml is, and what all the sections do. Looking at the gtk projects Gir.toml reveals a whole bunch of extra things in [[object]] sections.

Thanks for working on this!

File: README.md [hunk 5]
```diff
@@ -1,8 +1,29 @@
...
+cargo run --release -c YourGirFile.toml -d ../gir-files -m sys -o the-output-directory-sys
 ```
-where `gir-files` contains the [GIR definitions](https://github.com/gkoz/gir-files).
-The generated files will be placed in `gtk`
+
+The generated files will be placed in `the-output-directory-sys`. You now have the sys part of your binding!
+
+## Generate the Rust-user API level
+
+You'll now have to write your own GIR file (take a look to [gtk/Gir.toml](https://github.com/gtk-rs/gtk/blob/master/Gir.toml)).
```
Comment (EPashkin) [inline README.md]: No same part for ffi?

**Comment by EPashkin** [comment]
@GuillaumeGomez you will planing to add link to ffi config example or prefer merge as is?

**Comment by GuillaumeGomez** [comment]
Sorry, I didn't update yet. I intend to add it.

**Comment by EPashkin** [comment]
Ok, thanks for doing this part. I bad with writing.

**Comment by GuillaumeGomez** [comment]
No problem, thanks a lot for reviewing and working on gtk-rs!

**Comment by GuillaumeGomez** [comment]
@EPashkin: I actually didn't understand your comment. To what ffi config file are you referring to?

**Comment by EPashkin** [comment]
I mean add line like "You'll now have to write your own GIR file (link to sys gir-gtk.toml)" in FFI block
before or IMHO better instead "When you're ready,  let's generate the FFI part" (I don't understand what "Don't forget to download the gir files as well" mean as it was done in previous paragraph).

**Comment by EPashkin** [comment]
Ops, I forgot insert link: https://github.com/gtk-rs/sys/blob/master/conf/gir-gtk.toml

File: README.md (hunk 5 above)
Comment (EPashkin) [inline README.md]: May be need clarify that it different config

File: README.md [hunk 6]
```diff
@@ -1,8 +1,33 @@
...
+Then you need to write a TOML file you'll pass to gir (you can take a look to [gtk-rs/sys/gir-gtk.toml](https://github.com/gtk-rs/sys/blob/master/conf/gir-gtk.toml) to see an example).
+
+*This part needs more information about this toml file*
+
+When you're ready, let's generate the FFI part:
 
-Running:
 ```shell
-cargo run --release -- -c Gir_Gtk.toml -d ../gir-files -o gtk
+cargo run --release -c YourGirFile.toml -d ../gir-files -m sys -o the-output-directory-sys
```
Comment (EPashkin) [inline README.md]: YourGirFile.toml has minor conflict with "Then you need to write a TOML file" before.
Also API level says "You'll now have to write your own GIR file"

**Comment by EPashkin** [comment]
All look good for me.

**Comment by EPashkin** [comment]
Ready for merge?

**Comment by GuillaumeGomez** [comment]
Great, thanks! I merge then!

But we really need to add explanations on how to write/generate the TOML files. I think you are the one who knows about it the most. If you have difficulties on writing, just send me a very short set of "instructions" on how to do it and I'll improve the writing. I have absolutely no issue with that.

Then we'll need to start explaining a bit how gir code is working, it might help attracting new contributors who would try to add automatic binding generation for their own gnome/gir-compatible crate.

**Comment by EPashkin** [comment]
I think about it.

**Comment by GuillaumeGomez** [comment]
Great, thanks!

**Pull request merged**